code/modules/shuttle/navigation_computer.dm

```diff
@@ -0,0 +1,242 @@
+/obj/machinery/computer/camera_advanced/shuttle_docker
+	name = "navigation computer"
+	desc = "Used to designate a precise transit location for a spacecraft."
+	z_lock = ZLEVEL_STATION
+	jump_action = null
+	var/datum/action/innate/shuttledocker_rotate/rotate_action = new
+	var/datum/action/innate/shuttledocker_place/place_action = new
+	var/shuttleId = ""
+	var/shuttlePortId = ""
+	var/shuttlePortName = ""
+	var/list/jumpto_ports = list() //hashset of ports to jump to and ignore for collision purposes
+	var/list/blacklisted_turfs
+	var/obj/docking_port/stationary/my_port
+	var/view_range = 7
+	var/x_offset = 0
+	var/y_offset = 0
+	var/space_turfs_only = TRUE
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/GrantActions(mob/living/user)
+	if(jumpto_ports.len)
+		jump_action = new /datum/action/innate/camera_jump/shuttle_docker
+	..()
+
+	if(rotate_action)
+		rotate_action.target = user
+		rotate_action.Grant(user)
+		actions += rotate_action
+
+	if(place_action)
+		place_action.target = user
+		place_action.Grant(user)
+		actions += place_action
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/CreateEye()
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
+	if(!M)
+		return
+	eyeobj = new /mob/camera/aiEye/remote/shuttle_docker()
+	var/mob/camera/aiEye/remote/shuttle_docker/the_eye = eyeobj
+	the_eye.origin = src
+	the_eye.dir = M.dir
+	var/area/A = get_area(M)
+	if(!A)
+		return
+	var/turf/origin = locate(M.x + x_offset, M.y + y_offset, M.z)
+	for(var/turf/T in A)
+		if(T.z != origin.z)
+			continue
+		var/image/I = image('icons/effects/alphacolors.dmi', origin, "red")
+		I.layer = ABOVE_NORMAL_TURF_LAYER
+		I.plane = 0
+		I.mouse_opacity = 0
+		var/x_off = T.x - origin.x
+		var/y_off = T.y - origin.y
+		I.pixel_x = x_off * 32
+		I.pixel_y = y_off * 32
+		the_eye.placement_images[I] = list(x_off, y_off)
+	generateBlacklistedTurfs()
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/give_eye_control(mob/user)
+	..()
+	if(user && user.client)
```

Comment:
Mobs are immediately harddelled, and even if they were not, user.client would be null.